test/show.jl

```diff
@@ -311,6 +314,33 @@ end
     ├─────┼───────────┤
     │ 1   │ 1.0       │
     │ 2   │ missing   │"""
+
+    # date types
+    df = DataFrame(a = Date(2020, 2, 11), b = DateTime(2020, 2, 11, 15), c = Day(1))
+    @test sprint(show, df) == """
+    1×3 DataFrame
+    │ Row │ a          │ b                   │ c     │
+    │     │ Date       │ DateTime            │ Day   │
+    ├─────┼────────────┼─────────────────────┼───────┤
+    │ 1   │ 2020-02-11 │ 2020-02-11T15:00:00 │ 1 day │"""
+
+    # Irrational
+    df = DataFrame(a = π)
+    if VERSION < v"1.2.0-DEV.276"
```

Comment:
This doesn't looks like an acceptable result given how verbose is, and we still claim to support 1.0. It might be OK if that only affects `Rational` as it's unlikely to be stored in a data frame. But is there a risk that it affects other custom types (notably from packages, like Unitful.jl)?

Reply:
It should be noted that this is just mirroring how DataFrames and Julia already work. For example here is the behaviour of DataFrames v0.20.0 with various versions of Julia:

Julia 1.0.5
```julia
julia> using DataFrames; DataFrame(a = π)
1×1 DataFrame
│ Row │ a                      │
│     │ Irrational{:π}         │
├─────┼────────────────────────┤
│ 1   │ π = 3.1415926535897... │
```

Julia 1.1.1
```
julia> using DataFrames; DataFrame(a = π)
1×1 DataFrame
│ Row │ a                      │
│     │ Irrational{:π}         │
├─────┼────────────────────────┤
│ 1   │ π = 3.1415926535897... │
```

Julia 1.2.0
```julia
julia> using DataFrames; DataFrame(a = π)
1×1 DataFrame
│ Row │ a         │
│     │ Irration… │
├─────┼───────────┤
│ 1   │ π         │
```

Julia 1.3.1
```julia
julia> using DataFrames; DataFrame(a = π)
1×1 DataFrame
│ Row │ a         │
│     │ Irration… │
├─────┼───────────┤
│ 1   │ π         │
```

Reply:
Ah OK. Clearly nobody cares or we would have got reports about it. :-)